subsys/debug/tracing/CMakeLists.txt

```diff
@@ -1,27 +1,42 @@
 # SPDX-License-Identifier: Apache-2.0
 
 if(CONFIG_SEGGER_SYSTEMVIEW OR CONFIG_TRACING_CPU_STATS)
-  zephyr_library()
 
-  zephyr_library_sources_ifdef(
+  zephyr_sources_ifdef(
     CONFIG_SEGGER_SYSTEMVIEW
     sysview_config.c
     sysview.c
     )
 
-  zephyr_library_sources_ifdef(
+  zephyr_sources_ifdef(
     CONFIG_TRACING_CPU_STATS
     cpu_stats.c
     )
-
-  if(CONFIG_TRACING)
-    zephyr_library_include_directories(
-      ${ZEPHYR_BASE}/kernel/include
-      ${ZEPHYR_BASE}/arch/${ARCH}/include
-      )
-  endif()
 endif()
 
+zephyr_sources_ifdef(
+  CONFIG_TRACING
+  tracing_buffer.c
+  tracing_core.c
+  tracing_format.c
```

Comment:
CONFIG_TRACING is the most generic tracing kconfig. It will be defined for Segger and CTF and this.
However do we want to force the files upon people using Segger or CTF?
Solution: Specialize this kconfig

Reply:
@mped-oticon is CONFIG_TRACING_GENERIC good for you?

Reply:
@wentongwu yep